scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,523 @@
-    { x = -261, y = 8, z = 400 },
-    { x = -257, y = 12, z = 417 },
-    { x = -219, y = 16, z = 420 },
-    { x = -220, z = 448 },
-    { x = -212, z = 459 },
-    { x = -130, z = 458 },
-    { x = -118, z = 420 },
-    { x = -131, z = 381 },
-    { x = -212, z = 380 },
-    { x = -220, z = 392 },
-    { x = -219, z = 420 },
-    { x = -257, y = 12, z = 417 },
-    { x = -261, y = 8, z = 400 },
-    { x = -256, y = 4, z = 381 },
-    { x = -238, y = 0, z = 380 },
-    { x = -214, z = 381 },
-    { x = -209, z = 387 },
-    { x = -202, z = 391 },
+    { x = 219, y = 16, z =  387},
+    { x = 215, y = 16, z = 380 },
+    { x = 211, y = 16, z = 380 },
+    { x = 142, y = 16, z = 380 },
+    { x = 139, y = 16, z = 384 },
+    { x = 139, y = 16, z = 419 },
+    { x = 145, y = 16, z = 427 },
+    { x = 173, y = 16, z = 458 },
+    { x = 178, y = 16, z = 460 },
+    { x = 210, y = 16, z = 460 },
+    { x = 215, y = 16, z = 457 },
+    { x = 219, y = 16, z = 452 },
+    { x = 219, y = 16, z = 442 },
+    { x = 219, y = 16, z = 420 },
+    { x = 240, y = 16, z = 419 },
+    { x = 258, y = 12, z = 417 },
+    { x = 259, y = 4, z = 384 },
+    { x = 254, y = 3, z = 380 },
+    { x = 239, y = 0, z = 379 },
+    { x = 214, y = 0, z = 380 },
+    { x = 208, y = 0, z = 386 },
+    { x = 208, y = 0, z = 398 },
+    { x = 206, y = 0, z = 405 },
+    { x = 198, y = 0, z = 409 },
+    { x = 191, y = 0, z = 405 },
+    { x = 192, y = 0, z = 395 },
+    { x = 193, y = 0, z = 389 },
+    { x = 187, y = 0, z = 382 },
+    { x = 184, y = 0, z = 380 },
+    { x = 164, y = 0, z = 380 },
 }
 
+    -- path from the first yellow door to the middle yellow door
+local firstYellowToMiddleYellow =
+{
+    { x = 156, y = 0, z = 379 },-- first yellow door
+    { x = 144, y = 0, z = 379},
+    { x = 133, y = 0, z = 380 },
+    { x = 129, y = 0, z = 384 },
+    { x = 124, y = 0, z = 399 },
+    { x = 124, y = 0, z = 410 },
+    { x = 117, y = 0, z = 424 },
+    { x = 114, y = 0, z = 439 },
+    { x = 111, y = 0, z = 451 },
+    { x = 109, y = 0, z = 456 },
+    { x = 106, y = 0, z = 458 },
+    { x = 100, y = 0, z = 460 },
+    { x = 94, y = 0, z = 459 },
+    { x = 82, y = 0, z = 459 },-- Middle yellow door
+}
+
+    -- path from the yellow middle door to the middle blue door
+local middleYellowToMiddleBlue =
+{
+    { x = 77, y = 0, z = 460 },
+    { x = 71, y = 0, z = 459 },
+    { x = 66, y = 0, z = 458 },
+    { x = 63, y = 0, z = 456 },
+    { x = 60, y = 0, z = 450 },
+    { x = 60, y = 0, z = 443 },
+    { x = 60, y = 0, z = 419 },
+    { x = 40, y = 0, z = 419 },
+    { x = 21, y = 0, z = 419 },
+    { x = -9, y = 0, z = 420 },
+    { x = -59, y = 0, z = 420 },
+    { x = -60, y = 0, z = 433 },
+    { x = -60, y = 0, z = 450 },
+    { x = -62, y = 0, z = 455 },
+    { x = -67, y = 0, z = 458 },
+    { x = -75, y = 0, z = 460 },
+    { x = -76, y = 0, z = 460 },
+}
+
+    -- Path from the middle blue door to the first blue door
+local middleBlueToFirstBlue =
```

Comment:
The naming of these is inconistant, you're mixing up "position" and "colour" in the naming of these. This is also presumably just the reverse of `firstBluetoBlueMiddle`. See my comment somewhere below about using the reverse flag.

Reply:
Indeed, the reverse will not work for the basement loops as Zip goes in the same direction regardless. The reverse flag would have it go counter to that. For the sake of uniformity, I opted to not include those for the middle doors as I thought it might be confusing.

Reply:
Helping Hook out with this - So will comment on some of this on his behalf.
With the removal of excess local vars and using an indexed table to determine pathing, this should resolve most of this I believe.  I am still not a large fan of the naming convention myself, but I kept things as close to his as I could.

If he needs to do extra modifications on names, I will leave that to him for now.